keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/TimeWindowConstraintEvaluator.kt

```
@@ -0,0 +1,203 @@
+    }
+
+    val days = mutableSetOf<String>()
+    val trimmed = dayConfig.replace(whiteSpace, "")
+      .toLowerCase()
+    val elements = trimmed.split(",")
+
+    elements.forEach {
+      when {
+        it.isDay() -> days.add(it)
+        it.isDayRange() -> days.addAll(it.dayRange())
+        else -> throw InvalidConstraintException(
+          "Invalid time-window constraint ($it) on deliveryConfig: ${deliveryConfig.name}, " +
+            "application: ${deliveryConfig.application}, environment: $targetEnvironment")
+      }
+    }
+
+    return days
+  }
+
+  private fun String.isInt(): Boolean = this.matches(intOnly)
+
+  private fun String.isIntRange(): Boolean = this.matches(intRange)
+
+  private fun String.hourRange(): Set<Int> {
+    val hours = this.split("-")
+    return IntRange(hours[0].toInt(), hours[1].toInt())
+      .toSet()
+  }
+
+  private fun String.isDay(): Boolean = daysOfWeek.contains(this)
+
+  private fun String.isDayRange(): Boolean {
+    val days = this.split("-")
+    if (days.size != 2) {
+      return false
+    }
+
+    return daysOfWeek.contains(days[0]) && daysOfWeek.contains(days[1])
+  }
+
+  private fun String.dayRange(): Set<String> {
+    /**
+     * Convert Mon-Fri or Monday-Friday to [DayOfWeek] integers to compute
+     * a range, then back to a set of individual days that today can be
+     * matched against.
+     */
+    val days = this.split("-").map { it.capitalize() }
+    val day1 = fullDayFormatter.parseUnresolved(days[0], ParsePosition(0))
+      ?: shortDayFormatter.parseUnresolved(days[0], ParsePosition(0))
+      ?: throw InvalidConstraintException("Failed parsing day range $this")
+    val day2 = fullDayFormatter.parseUnresolved(days[1], ParsePosition(0))
+      ?: shortDayFormatter.parseUnresolved(days[1], ParsePosition(0))
+      ?: throw InvalidConstraintException("Failed parsing day range $this")
+
+    val intRange = IntRange(DayOfWeek.from(day1).value, DayOfWeek.from(day2).value)
```

What would happen here if you specify something backwards like `thu-mon`?

---

Probably not what the user would want.

---

Should we throw an `InvalidConstraintException` in that case, then?

---

Should we throw an `InvalidConstraintException` in that case, then?

---

Updated the behavior so wrap around hours and days work as one would reasonably expect.